theano/gof/graph.py

```diff
@@ -406,8 +406,32 @@ def __str__(self):
         else:
             return "<%s>" % str(self.type)
 
-    def __repr__(self):
-        return str(self)
+    def __str_test_value__(self):
+        """Return a repr of the test value.
+
+        Return a printable representation of the test value. It can be
+        overridden by classes with non printable test_value to provide a
+        suitable representation of the test_value.
+        """
+        try:
+            return repr(theano.gof.op.get_test_value(self))
+        except:
+            raise
+
+    def __repr__(self, firstPass=True):
```

Comment:
I think it is the __str__ that must be redefined, not __repr__. Is there a reason you did the reverse?

Also, never make an try: ... except:

Always as least catch Exception. Here, catch only AttributeError